cql/SupportingData.js

```diff
@@ -0,0 +1,119 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+const xml2js = require('xml2js');
+const fs = require('fs');
+
+const myArgs = process.argv.slice(2);
+
+const xml = fs.readFileSync(myArgs[0], 'utf8');
+
+const stringToIsoDate = (value) => {
+  if (value === '') {
+    return null;
+  }
+  if (value.length !== 8) {
+    throw new Error(`Invalid date ${value}`);
+  }
+  return `${value.substring(0, 4)}-${value.substring(4, 6)}-${value.substring(6)}`;
+}
+
+const stringToQuantityUnit = (quantityUnit) => {
+  if (quantityUnit === '') {
+    return '';
+  }
+  switch (quantityUnit) {
+    case 'day': return 'day';
+    case 'week': return 'week';
+    case 'month': return 'month';
+    case 'year': return 'year';
+    case 'days': return 'days';
+    case 'weeks': return 'weeks';
+    case 'months': return 'months';
+    case 'years': return 'years';
```

Comment:
It turns out the CQL spec date functions don't use the same units as FHIR quantity units for dates. Since the CDC xml uses the same units as CQL (just by coincidence) I changed it to use those units.  Updated to use your suggested logic but changed to throw an error if other unit is used, so it doesn't fail silently.

Reply:
Throwing an error so it doesn't fail silently sounds good to me 👍 .  Looking through the spec it seems like CQL (e.g. `week`) or UCUM form (e.g. `weeks`) are allowed by date operations which explains why this didn't cause errors previously (at least I'm guessing that's why)


[Note that the quantity units may be specified in singular, plural, or UCUM form](https://cql.hl7.org/09-b-cqlreference.html#add-1)